middleware/grpc_logging.go

```diff
@@ -16,6 +17,11 @@ const (
 	errorKey = "err"
 )
 
+// If an error implements this interface, it will get called and GRPCServerLog will do nothing.
+type CustomLog interface {
+	LogOperation(ctx context.Context, _ logging.Interface, method string, duration time.Duration)
```

Comment:
Opinionated nit, I'd call this:
```suggestion
type Observer interface {
	Observe(ctx context.Context, _ logging.Interface, method string, duration time.Duration)
```

Why:
- `LogOperation` implies caller has to log an _operation_, but this is not something we require: they may or may not log it, we just want them to be aware of it, i.e. `Observe` it.
- `Observer` because that's idiomatic golang for single-method interfaces.

Reply:
OK I like `Observe` better, but does it convey that we no longer do the regular logging operation?